algo_lib/src/io/output.rs

```diff
@@ -1,44 +1,85 @@
 use std::io::Write;
 
-pub struct Output {
-    output: Box<dyn Write>,
+#[derive(Copy, Clone)]
+pub enum BoolOutput {
+    YesNo,
+    YesNoCaps,
+    PossibleImpossible,
+    Custom(&'static str, &'static str),
+}
+
+impl BoolOutput {
+    pub fn output(&self, output: &mut Output, val: bool) {
+        (if val { self.yes() } else { self.no() }).write(output);
+    }
+
+    fn yes(&self) -> &str {
+        match self {
+            BoolOutput::YesNo => "Yes",
+            BoolOutput::YesNoCaps => "YES",
+            BoolOutput::PossibleImpossible => "Possible",
+            BoolOutput::Custom(yes, _) => yes,
+        }
+    }
+
+    fn no(&self) -> &str {
+        match self {
+            BoolOutput::YesNo => "No",
+            BoolOutput::YesNoCaps => "NO",
+            BoolOutput::PossibleImpossible => "Impossible",
+            BoolOutput::Custom(_, no) => no,
+        }
+    }
+}
+
+pub struct Output<'s> {
+    output: &'s mut dyn Write,
     buf: Vec<u8>,
     at: usize,
     auto_flush: bool,
+    bool_output: BoolOutput,
```

Comment:
To me, it seems like `bool_output` is a little bit different abstraction level -- currently, `struct Output` is just some struct to print something to files/stdout, which is possible to reuse for other projects, not necessarily competitive programming related. 

But `bool_output` is a very niche thing related to the contests.

Reply:
I agree, I just blindly copied it from my own library. Will get rid of this.